tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovPrinter.java

```diff
@@ -132,16 +131,12 @@ private void printFNHLine(SourceFileCoverage sourceFile) throws IOException {
   // BRDA:<line number>,<block number>,<branch number>,<taken>
   private void printBRDALines(SourceFileCoverage sourceFile) throws IOException {
     for (BranchCoverage branch : sourceFile.getAllBranches()) {
-      if (branch.blockNumber().isEmpty() || branch.branchNumber().isEmpty()) {
-        // We skip printing this as a BRDA line and print it later as a BA line.
-        continue;
-      }
       bufferedWriter.write(Constants.BRDA_MARKER);
       bufferedWriter.write(Integer.toString(branch.lineNumber()));
       bufferedWriter.write(Constants.DELIMITER);
-      bufferedWriter.write(branch.blockNumber());
+      bufferedWriter.write(branch.blockNumber().isEmpty() ? "0" : branch.blockNumber());
```

Comment:
is the printBALines method dead code after this change?

Reply:
Exactly. Everything about BA-lines is now obsolete. Do you know where the BA-lines came from? From Gcov/Lcov/GenInfo side I was not able to find something about, even not in the history.